nautobot_circuit_maintenance/jobs/site_search.py

```diff
@@ -0,0 +1,161 @@
+"""Site searching Job definition."""
+import collections
+from datetime import datetime, date
+from typing import NamedTuple
+
+from django.conf import settings
+
+from nautobot.extras.jobs import Job
+from nautobot.circuits.models import Circuit
+
+from nautobot_circuit_maintenance.models import CircuitMaintenance
+
+name = "Circuit Maintenance"  # pylint: disable=invalid-name
+PLUGIN_SETTINGS = settings.PLUGINS_CONFIG.get("nautobot_circuit_maintenance", {})
+CIRCUIT_MAINTENANCE_TAG_COLOR = "Purple"
+
+
+class Range(NamedTuple):
+    """Range class for use within the check for overlap.
+
+    Args:
+        NamedTuple (NamedTupe): Named Tuple set up
+    """
+
+    start: datetime
+    end: datetime
+
+
+def check_for_overlap(record1: CircuitMaintenance, record2: CircuitMaintenance):
+    """Checks for the overlap of two circuit maintenance records.
+
+    Args:
+        record1 (CircuitMaintenance): First maintenance record
+        record2 (CircuitMaintenance): Second maintenance record
+
+    Returns:
+        bool: Result of there is overlap
+    """
+    # Build a couple of ranges
+    range1 = Range(start=record1.start_time, end=record1.end_time)
+    range2 = Range(start=record2.start_time, end=record2.end_time)
+
+    # Determine the latest start and the earliest end to determine overlap
+    latest_start = max(range1.start, range2.start)
+    earliest_end = min(range1.end, range2.end)
+
+    # Get the delta, compare versus 0 (negative number is no overlap)
+    delta = (earliest_end - latest_start).days + 1
+
+    # If delta is negative there is no overlap, return false
+    overlap = max(0, delta)
+
+    return overlap != 0
```

Comment:
I believe a more explicit logic should be possible...

Reply:
What do we mean by this?